src/sidebar-calendar-loader.ts

```diff
@@ -0,0 +1,54 @@
+let sidebarCalendarOffset: number = 0;
+
+function loadSidebarCalendar(){
+    loadSidebarCalendarDateLabels();
+    loadSidebarCalendarDate();
+}
+
+function loadSidebarCalendarDateLabels(){
+    const currentMonth: Date = new Date();
+    currentMonth.setMonth(currentMonth.getMonth() + sidebarCalendarOffset);
+    const calendarModule: HTMLElement = document.getElementsByClassName("calendar-module")[0] as HTMLElement;
+    const calendarModuleDateButtons: HTMLElement[] = Array.from(calendarModule.getElementsByClassName("calendar-module__day-cell")) as HTMLElement[];
+
+    const firstDay: Date = new Date(currentMonth).getFirstDayOfMonth();
+    const lastDay: Date = new Date(currentMonth).getLastDayOfMonth();
+    const firstDayOfMonthWeek: Date = firstDay.getFirstDayOfWeek();
+    const lastDayOfMonthWeek: Date = new Date(lastDay).getLastDayOfWeek();
+
+    calendarModuleDateButtons.forEach(button => {
+        button.remove();
+    });
+    for (let i: Date = new Date(firstDayOfMonthWeek);
+         i.toDateString() !== lastDayOfMonthWeek.addDays(1).toDateString();
+         i.setDate(i.getDate() + 1)) {
+
+        const calendarButton: HTMLElement = document.createElement("button");
+        calendarButton.innerText = String(i.getDate());
+        calendarButton.classList.add("calendar-module__day-cell", "button-backgroundless", "button-borderless");
+        calendarButton.setAttribute("data-sidebar-calendar-date", i.getTime().toString());
+        calendarButton.addEventListener("click", async () => {
+            const buttonDate: number = Number(calendarButton.getAttribute("data-sidebar-calendar-date"));
+            weekOffset = weekOffsetCalc(new Date(), new Date(buttonDate));
+            await reloadWeekView();
+        });
+
+        if (i.toYearMonthString() !== currentMonth.toYearMonthString())
+            calendarButton.classList.add("calendar-module__day-cell-not-current");
+
+        calendarModule.appendChild(calendarButton);
+    }
+}
+
+function loadSidebarCalendarDate() {
+    const headerDate: HTMLElement = document.getElementsByClassName("calendar-module-header__date")[0] as HTMLElement;
+    const currentMonth: Date = new Date();
+    currentMonth.setMonth(currentMonth.getMonth() + sidebarCalendarOffset);
+    headerDate.innerText = `${currentMonth.getFullYear()} ${MONTHS[currentMonth.getMonth()]}`
```

Comment:
separation of concerns: formatting, DOM manipulation, business logic. More work, but easier to maintain and scale.  First indication is when functions get too long. My preference is to write code in a black box concept, where functions have no side effects (here it would be DOM manipulation, in other cases network request, cache access etc). Of course, it's not possible for function to not have side effects, so we tend to bubble this logic up and do side effects as far as possible from deeply nested function calls (ideally at the beginning).